skynet/auth/openai.py

```diff
@@ -0,0 +1,29 @@
+import yaml
+
+from skynet.env import openai_credentials_file
+from skynet.modules.file_watcher import FileWatcher
+
+credentials = dict()
+
+
+def open_yaml(file_path):
+    try:
+        with open(file_path, 'r') as stream:
+            global credentials
+            credentials = yaml.safe_load(stream)['customer_credentials']
+    except Exception as e:
+        raise RuntimeError(f'Error loading credentials file: {e}')
+
+
+def setup_credentials():
+    if openai_credentials_file:
+        open_yaml(openai_credentials_file)
+        file_watcher = FileWatcher(openai_credentials_file, lambda: open_yaml(openai_credentials_file))
+
+        file_watcher.start()
```

Comment:
When are we going to stop this?

Reply:
when the process stops. Is there other time when we'd want it to stop

Reply:
Not really, in general I prefer explicit vs implicit. No big deal though.